### DISCUSSION TITLE
When is F# code analysis coming in visual studio

### DISCUSSION THREAD

**Discussion by Xyncgas** [dotnet/fsharp#16088] [Q&A] [answered]
C# has analysis for removing unused using statement, etc.
F# has nothing yet so I'm wondering

**Comment by vzarytovskii** [comment, marked as answer]
Visual Studio has some, analyzers sdk will be discussed in future.

**Comment by smoothdeveloper** [comment]
@vzarytovskii, is there a place where the possible designs are being discussed, reviewed, etc.

The analyzer SDK could be designed in a way where the analyzer compilation (against some version of FCS) would export the signature metadata of all members accessed by the analyzer, in a fine grained way matching the IL.

On the host side, it would have ability to host several versions of FCS, and pick the most recent one that covers all the symbols used by a particular analyzer (or some other strategy).

It would remove the need to keep some form of binary compability when some breaking changes to the AST are necessary.

**Comment by vzarytovskii** [comment]
> @vzarytovskii, is there a place where the possible designs are being discussed, reviewed, etc.
> 
> 
> 
> The analyzer SDK could be designed in a way where the analyzer compilation (against some version of FCS) would export the signature metadata of all members accessed by the analyzer, in a fine grained way matching the IL.
> 
> 
> 
> On the host side, it would have ability to host several versions of FCS, and pick the most recent one that covers all the symbols used by a particular analyzer (or some other strategy).
> 
> 
> 
> It would remove the need to keep some form of binary compability when some breaking changes to the AST are necessary.

Nothing at the moment.
